lsp/server/src/server.ts

```
@@ -33,93 +33,136 @@ const documents: TextDocuments<TextDocument> = new TextDocuments(TextDocument);
-
+let messDef: string;
+let allowConfigurationChanges = false;
 let startedSA4U_Z3 = false;
-connection.onInitialize((params: InitializeParams) => {
 
-	const capabilities = params.capabilities;
+const execAsync = promisify(exec);
 
-	// Does the client support the `workspace/configuration` request?
-	// If not, we fall back using global settings.
-	hasConfigurationCapability = !!(
-		capabilities.workspace && !!capabilities.workspace.configuration
-	);
-	hasWorkspaceFolderCapability = !!(
-		capabilities.workspace && !!capabilities.workspace.workspaceFolders
-	);
+class SA4UConfig {
+	compilationDir: string
+	constructor(compilationDir: string) {
+		this.compilationDir = '/src/' + compilationDir;
+	}
+}
 
-	const result: InitializeResult = {
-		capabilities: {
-			codeActionProvider: true,
-			textDocumentSync: TextDocumentSyncKind.Incremental,
-			executeCommandProvider: {
-				commands: ['sa4u.fix']
-			}
+class SA4UIgnoredFiles {
+	ignoreFilesCommand: string
+	constructor(ignoreFilesArray: string[]) {
+		if (ignoreFilesArray.length != 1 || !(ignoreFilesArray[0].match(''))) {
+			this.ignoreFilesCommand = '-i '.concat(ignoreFilesArray.join(' -i '));
+		} else {
+			this.ignoreFilesCommand = '';
 		}
-	};
-	if (hasWorkspaceFolderCapability) {
-		result.capabilities.workspace = {
-			workspaceFolders: {
-				supported: true
-			}
-		};
-	}
-	return result;
-});
+	} 
+}
 
-connection.onInitialized(() => {
-	
-	if (hasConfigurationCapability) {
-		// Register for all configuration changes.
-		connection.client.register(DidChangeConfigurationNotification.type, undefined);
-		connection.workspace.getConfiguration('SA4U').then((value) => {
-			if (value.messageDefinition) {
-				messDef = value.messageDefinition;
-			}
-		});
-	}
-	if (hasWorkspaceFolderCapability) {
-		connection.workspace.onDidChangeWorkspaceFolders(_event => {
-			connection.console.log('Workspace folder change event received.');
+async function checkForDockerRunningAndInstalled() {
```

Let's annotate the types of all top-level functions.

---

Also, we should probably avoid returning a union type in this situation. Instead, let's raise an exception.

---

All top-level function return types are now annotated. Threw an exception here.